frontend/stmt_inferrer.py

```diff
@@ -48,95 +49,96 @@ def _infer_assignment_target(target, context, value_type, lineno):
 
     if isinstance(node.target, ast.Name):
-        z3_types.type_solver.add(axioms.assignment(target_type, result_type),
-                                 fail_message="Augmented assignment in line {}".format(node.lineno))
+        solver.add(axioms.assignment(target_type, result_type, solver.z3_types),
+                   fail_message="Augmented assignment in line {}".format(node.lineno))
     elif isinstance(node.target, ast.Subscript):
-        indexed_type = expr.infer(node.target.value, context)
+        indexed_type = expr.infer(node.target.value, context, solver)
         if isinstance(node.target.slice, ast.Index):
-            index_type = expr.infer(node.target.slice.value, context)
-            z3_types.type_solver.add(axioms.index_assignment(indexed_type, index_type, result_type),
-                                     fail_message="Subscript augmented assignment in line {}".format(node.lineno))
+            index_type = expr.infer(node.target.slice.value, context, solver)
+            solver.add(axioms.index_assignment(indexed_type, index_type, result_type, solver.z3_types),
+                       fail_message="Subscript augmented assignment in line {}".format(node.lineno))
         else:
             lower_type = upper_type = step_type = z3_types.Int
             if node.target.slice.lower:
-                lower_type = expr.infer(node.target.slice.lower, context)
+                lower_type = expr.infer(node.target.slice.lower, context, solver)
             if node.target.slice.upper:
-                upper_type = expr.infer(node.target.slice.upper, context)
+                upper_type = expr.infer(node.target.slice.upper, context, solver)
             if node.target.slice.step:
-                step_type = expr.infer(node.target.slice.step, context)
-            z3_types.type_solver.add(
-                axioms.slice_assignment(lower_type, upper_type, step_type, indexed_type, result_type),
-                fail_message="Slice augmented assignment in line {}".format(node.lineno)
-            )
+                step_type = expr.infer(node.target.slice.step, context, solver)
+            solver.add(axioms.slice_assignment(lower_type, upper_type, step_type,
+                                               indexed_type, result_type, solver.z3_types),
+                       fail_message="Slice augmented assignment in line {}".format(node.lineno))
 
     elif isinstance(node.target, ast.Attribute):
-        # TODO: Implement after classes inference
-        pass
-    return z3_types.zNone
+        solver.add(axioms.assignment(target_type, value_type, solver.z3_types),
+                   fail_message="Augmented attribute assignment in line {}".format(node.lineno))
+    return solver.z3_types.none
 
 
 # noinspection PyUnresolvedReferences
-def _delete_element(target, context, lineno):
+def _delete_element(target, context, lineno, solver):
```

Comment:
I don't think we want to support attribute deletion. You can't really have a nominal type system where instances of some classes sometimes have some attribute but not always. So I'd just add a case where we output an error if an attribute is deleted. Actually, you should have that case anyway (with a NotImplementedError or something like that).